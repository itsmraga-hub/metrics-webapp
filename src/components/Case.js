import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { fetchsingleCountryInfo } from '../redux/covidcases';
import style from './css/Case.module.css';

const Case = (props) => {
  const coun = props;
  const { country, i } = coun;

  // eslint-disable-next-line camelcase
  const { abbreviation, id, life_expectancy } = country;

  const dispatch = useDispatch();

  const handleClick = (e) => {
    dispatch(fetchsingleCountryInfo(e.target.id));
  };

  return (
    <li className={`${style.Container} ${i % 4 === 0 || (i - 1) % 4 === 0 ? style.even : style.odd}`}>
      <Link to={{ pathname: `/country/${id}`, state: {} }}>
        <button type="button" id={id} onClick={handleClick}>
          <i className="fa-solid fa-circle-chevron-right fa-2x" id={id} />
        </button>
      </Link>
      <div>
        <div>
          <p>{abbreviation}</p>
          <span>{id}</span>
        </div>
        <p>
          Life expectancy:
          {' '}
          {/* eslint-disable-next-line camelcase */}
          {life_expectancy}
        </p>
      </div>
    </li>
  );
};

export default Case;
