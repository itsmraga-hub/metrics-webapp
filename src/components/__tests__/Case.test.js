import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../../redux/configureStore';

import Case from '../Case';

const coun = {
  country: {
    abbreviation: 'KE',
    id: 'Kenya',
    life_expectancy: '45',
  },
};
it('<Case /> Renders Properly', () => {
  const tree = renderer.create(
    <BrowserRouter>
      <Provider store={store}>
        <Case country={coun} />
      </Provider>
    </BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
