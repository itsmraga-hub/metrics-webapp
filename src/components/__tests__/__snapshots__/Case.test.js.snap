// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Case /> Renders Properly 1`] = `
<li
  className="Container odd"
>
  <a
    href="/country/undefined"
    onClick={[Function]}
  >
    <button
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa-solid fa-circle-chevron-right fa-2x"
      />
    </button>
  </a>
  <div>
    <div>
      <p />
      <span />
    </div>
    <p>
      Life expectancy:
       
    </p>
  </div>
</li>
`;
